components/concurrency_manager/src/key_handle.rs

```diff
@@ -39,14 +39,29 @@ impl KeyHandle {
     pub fn with_lock<T>(&self, f: impl FnOnce(&Option<Lock>) -> T) -> T {
         f(&*self.lock_store.lock())
     }
+
+    /// Set the LockTable that the KeyHandle is in.
+    ///
+    /// This method is not thread safe. Make sure that no other threads access
+    /// `table` at the same time.
+    pub(crate) unsafe fn set_table(&self, table: LockTable) {
+        *self.table.get() = Some(table);
+    }
```

Comment:
Must it be unsafe? Do we have any other way?
I don't mind it much though, just being afraid that the more unsafe code, the more unsafe 🤔

Reply:
I agree. I just think that code touching this unsafe is quite limiting now, so it's CURRENTLY easy to check its safety.
If it will be accessed in more places, then it's better to use a Mutex.